plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -97,6 +97,47 @@ static int init_put_payload(struct flb_kinesis *ctx, struct flush *buf,
     return -1;
 }
 
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+static void get_time_format_string(char *str, size_t maxsize,
+                      const char *time_key_format, const struct tm *timeptr,
+                      size_t len, struct flb_time *tms){
+    int millisecond_length = ((int)((ceil(log10(abs((uint64_t) tms->tm.tv_nsec / 1000000)))+1)*sizeof(char)));
```

Comment:
variable declaration like
int millisecond_length

should be at the top of your function but the definition should be afterwards like
milisecond_length = 1;